src/Qwiq.Core.Rest/QueryFolder.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+
+namespace Microsoft.Qwiq.Client.Rest
+{
+    internal class QueryFolder : Qwiq.QueryFolder
+    {
+        internal QueryFolder(QueryHierarchyItem queryFolder, Func<string, QueryHierarchyItem> folderExpansionFunc)
+            : base(
+                queryFolder.Id,
+                queryFolder.Name,
+                new QueryFolderCollection(() =>
+                {
+                    if (queryFolder.Children == null)
+                    {
+                        return Enumerable.Empty<IQueryFolder>();
+                    }
+                    
+                    return queryFolder
+                        .Children
+                        .Where(q => q?.IsFolder != null && q.IsFolder.Value)
```

Comment:
Couldn't this be shortened to `q?.IsFolder?.Value`?

Reply:
Nope, nullable types don't support safe navigation operator, would also rely on null being falsy.

Reply:
updated